src/microcosm.js

```diff
@@ -126,12 +131,17 @@ inherit(Microcosm, Emitter, {
     return this
   },
 
-  release (action) {
+  update () {
     let next = this.follower ? this.parent.state : this.recall(this.history.head)
 
-    if (next !== this.state) {
-      this.state = next
-      this._emit('change', next)
+    this.dirty = next !== this.state
+    this.state = next
+  },
+
+  release (action) {
+    if (this.dirty) {
+      this.dirty = false
+      this._emit('change', this.state)
     }
 
```

Comment:
I had this before, but removed it when I switched to the new reconciliation approach :-/